WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PageListViewModel.kt

```diff
@@ -240,21 +254,48 @@ class PageListViewModel @Inject constructor(
                     } else {
                         DEFAULT_INDENT
                     }
-                    PublishedPage(it.remoteId, it.title, it.date, labels, pageItemIndent,
-                            getFeaturedImageUrl(it.featuredImageId), actionsEnabled)
+                    val progressState = progressHelper.getProgressStateForPage(LocalId(it.pageId))
+
+                    PublishedPage(
+                            it.remoteId,
+                            it.title,
+                            it.date,
+                            labels,
+                            pageItemIndent,
+                            getFeaturedImageUrl(it.featuredImageId),
+                            actionsEnabled,
+                            progressState.first,
+                            progressState.second
+                    )
                 }
     }
 
-    private fun prepareScheduledPages(pages: List<PageModel>, actionsEnabled: Boolean): List<PageItem> {
+    private fun prepareScheduledPages(
+        pages: List<PageModel>,
+        actionsEnabled: Boolean
+    ): List<PageItem> {
         return pages.asSequence().groupBy { it.date.toFormattedDateString() }
-                .map { (date, results) -> listOf(Divider(date)) +
-                        results.map {
-                            val labels = mutableListOf<Int>()
-                            if (it.hasLocalChanges)
-                                labels.add(R.string.local_changes)
-
-                            ScheduledPage(it.remoteId, it.title, it.date, labels,
-                                getFeaturedImageUrl(it.featuredImageId), actionsEnabled) }
+                .map { (date, results) ->
+                    listOf(Divider(date)) +
+                            results.map {
+                                val labels = mutableListOf<Int>()
+                                if (it.hasLocalChanges)
+                                    labels.add(R.string.local_changes)
+
+                                val progressState = progressHelper.getProgressStateForPage(
```

Comment:
We might want to consider using this neat trick
`val (progressBarUiState, showOverlay)  = progressHelper.getProgressStateForPage(LocalId(it.pageId))`

Reply:
Cool idea!

Reply:
Resolved in 53cd03a